kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/BigQuerySinkTask.java

```diff
@@ -131,11 +132,23 @@ private PartitionedTableId getRecordTable(SinkRecord record) {
     // Dynamically update topicToBaseTableIds mapping. topicToBaseTableIds was used to be
     // constructed when connector starts hence new topic configuration needed connector to restart.
     // Dynamic update shall not require connector restart and shall compute table id in runtime.
-    if (!topicsToBaseTableIds.containsKey(record.topic())) {
-      TopicToTableResolver.updateTopicToTable(config, record.topic(), topicsToBaseTableIds);
+    TopicAndRecordName topicAndRecordName = new TopicAndRecordName(record.topic(), record.valueSchema().name());
+    if (!topicsToBaseTableIds.containsKey(topicAndRecordName)) {
+      TopicToTableResolver.updateTopicToTable(config, topicAndRecordName, topicsToBaseTableIds);
+      TableId tableId = topicsToBaseTableIds.get(topicAndRecordName);
+      if (bigQuery.getTable(tableId) == null) {
+        boolean createMissingTables = config.getBoolean(BigQuerySinkConfig.TABLE_CREATE_CONFIG);
+        if (createMissingTables) {
+          logger.info("Table {} does not exist. Attempting to create.", tableId.getTable());
+          schemaManager.createTable(tableId, topicAndRecordName);
+        } else {
+          throw new ConnectException(
```

Comment:
Do we need to specifically throw an exception here? Since if the table doesn't exist, bigquery will throw a corresponding exception while trying to insert records to that table.